### DISCUSSION TITLE
Did your plugin break?

### DISCUSSION THREAD

**Discussion by daemitus** [goatcorp/Dalamud#603] [Announcements]
Recently, as of 9/27, we made a change to how the plugin system pulls in dependencies for your plugin. This was mainly to solve the issue where you could not include a custom `FFXIVClientStructs` without being forced to merge the assembly. Now you can! As a byproduct however, Dalamud will not share any of it's own dependencies except in certain situations (i.e. Lumina). 

This has caused a plugin author bug to crop up, which I'm guilty of myself as well. If you have merged `FFXIVClientStructs.Generators.dll` your plugin will fail to load. This is due to a missing dependency on `Microsoft.CodeAnalysis` which was previously supplied because of Dalamud's scratchpad feature that has since been removed due to other issues. 

TLDR: Don't merge `FFXIVClientStructs.Generators.dll`. Dalamud should handle un-merged DLLs just fine now too.

If you insist on merging: here's some snippets you can use.

Using the ILRepack MSBuild task from NuGet.
```
 <Target Name="ILRepacker" AfterTargets="Build">
  <ItemGroup>
   <InputAssemblies Include="$(OutputPath)$(AssemblyName).dll" />
   <InputAssemblies Include="$(OutputPath)FFXIVClientStructs.dll" />
   <InputAssemblies Include="$(OutputPath)FFXIVClientStructs.Common.dll" />
  </ItemGroup>
  <ILRepack Parallel="true" Internalize="false" DebugInfo="true" InputAssemblies="@(InputAssemblies)"
            LibraryPath="$(Dalamud)" TargetKind="Dll" OutputFile="$(DevOut)$(AssemblyName).dll" />
 </Target>
```

Performing the command manually.
 ```
  <ItemGroup>
    <PackageReference Include="ILRepack" Version="2.0.18" GeneratePathProperty="true" />
  </ItemGroup>

  <Target Name="ILRepack" AfterTargets="Build">
    <ItemGroup>
      <InputAssemblies Include="$(TargetPath)" />
      <InputAssemblies Include="$(TargetDir)*.dll" Exclude="$(TargetPath);$(TargetDir)Scriban.dll;$(TargetDir)FFXIVClientStructs.Generators.dll" />
    </ItemGroup>
    <Exec Command="$(PkgILRepack)\tools\ILRepack.exe /lib:$(DalamudLibPath) /out:$(TargetDir)$(TargetFileName) @(InputAssemblies, ' ')" />
  </Target>
```
